### ISSUE TITLE
Unable to Provision ADLS storage : Error 403 in azure rm version 3.113

### ISSUE THREAD

**Issue by dev15119** [hashicorp/terraform-provider-azurerm#26785]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.9

### AzureRM Provider Version

3.113.0

### Affected Resource(s)/Data Source(s)

azurerm_3.113.0

### Terraform Configuration Files

```hcl
resource "azurerm_storage_account" "adls_accounts" {
 #for_each                      = { for idx, acc in var.adls_para : acc.name => idx }
 for_each                      = { for acc in var.adls_para : acc.name => acc }
 name                          = each.value.name
 resource_group_name           = each.value.resource_group_name
 location                      = each.value.location
 account_kind                  = each.value.account_kind
 account_tier                  = each.value.account_tier
 account_replication_type      = each.value.account_replication_type
 is_hns_enabled                = each.value.is_hns_enabled
 public_network_access_enabled = each.value.public_network_access_enabled
 access_tier                   = each.value.access_tier
 enable_https_traffic_only     = each.value.enable_https_traffic_only
 large_file_share_enabled      = each.value.large_file_share_enabled
 min_tls_version               = each.value.min_tls_version
 allow_nested_items_to_be_public = each.value.allow_nested_items_to_be_public
 default_to_oauth_authentication  = each.value.default_to_oauth_authentication
 shared_access_key_enabled    = each.value.shared_access_key_enabled
 infrastructure_encryption_enabled = each.value.infrastructure_encryption_enabled
 

 
blob_properties {
    delete_retention_policy {
      days = each.value.delete_retention_policy
    }
     container_delete_retention_policy {
    days = each.value.container_delete_retention_policy
  }
  }
 
 tags = each.value.tags
 lifecycle {
    ignore_changes = [
      tags
    ]
 }
 
 depends_on = [ azurerm_resource_group.adls ]
 
}
 
 
resource "azurerm_storage_data_lake_gen2_filesystem" "filesystems" {
  for_each =  {for acc in local.adls_file_system: "${acc.name}" => acc if acc.is_hns_enabled}
  name              = each.value.fsname
  storage_account_id = each.value.storage_account_id
 
 
  depends_on = [ azurerm_storage_account.adls_accounts ]
}
 

Provider 
terraform {
  required_version = ">= 1.2.2"
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
     }
    azuread = {
      source  = "hashicorp/azuread"
      # version = "~>2.45.0"
    }
    random = {
      source  = "hashicorp/random"
      version = "~>3.0"
     
    }
  }
```

### Debug Output/Panic Output

```shell
23T18:15:21.8785807Z [31m│[0m [0m[1m[31mError: [0m[0m[1mchecking for existence of existing File System "tst" in Account "*********" (IsEdgeZone false / ZoneName "" / Subdomain Type "dfs" / DomainSuffix "core.windows.net"): executing request: unexpected status 403 (403 This request is not authorized to perform this operation.) with EOF[0m
2024-07-23T18:15:21.8786681Z [31m│[0m [0m
```

### Expected Behaviour

This must provision without error

### Actual Behaviour

Failed during File system read

### Steps to Reproduce

Apply teh code with latest azurerm version 3.113.0

### Important Factoids

NO

### References

** https://github.com/Azure-Samples/azuresandbox/issues/29
Similar issue was shown in version before 3.0.0 **

**Comment by hqhqhqhqhqhqhqhqhqhqhq** [comment]
@dev15119 

Thanks for raising this!

I attempted to recreate this issue but was unable to reproduce the error. Could you please provide any additional details, such as any DEBUG log by setting the [TF_LOG](https://developer.hashicorp.com/terraform/internals/debugging) environment variable?

@manicminer Hi Tom! I saw your responses in the tagged issue. Would you happen to have any ideas on what's causing this? Thank you!

**Comment by dev15119** [comment]
When you will disable Public network & SAS key. You need to add use azure ad = true in provider in ADLS then this can be reproduced

**Comment by eehret** [comment]
I'm having a very similar problem, but in our case it's just a simple storage account for blob storage.  
It's also happening when shared key access is disabled.  We have turned on both `use_azuread_auth` and `storage_use_azuread`, to no avail.

During storage account creation:
```
│ Error: waiting for the Data Plane for Storage Account (Subscription: "<redacted>"
│ Resource Group Name: "lzroot-mgmt"
│ Storage Account Name: "stdtbazopsplfnsglogspr") to become available: waiting for the Blob Service to become available: polling failed: executing request: unexpected status 403 (403 Key based authentication is not permitted on this storage account.) with KeyBasedAuthenticationNotPermitted: Key based authentication is not permitted on this storage account.
│ RequestId:<redacted>
│ Time:2024-08-02T16:25:24.7707181Z
│
│   with module.storage_account_nsg_flow_logs.azurerm_storage_account.main,
│   on .terraform\modules\storage_account_nsg_flow_logs\storage.tf line 1, in resource "azurerm_storage_account" "main":
│    1: resource "azurerm_storage_account" "main" {
```

I was hoping that a workaround would be possible by pre-emptively assigning the 'Storage Blob Data Contributor` role to the resource group containing the storage account, but this did not make any difference.

As of right now it seems impossible to get a successful `terraform apply` when account key access has been disabled.

**Comment by dev15119** [comment]
I am already using this but this was not helpful.
In new version of azurerm 3.114.0 this seems to be solved

**Comment by dev15119** [comment]
solved in next version 3.111.0
